engines/dfc_provider/app/serializers/dfc_provider/offer_serializer.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+# Serializer used to render the DFC Offer from an OFN Product
+# into JSON-LD format based on DFC ontology
+module DfcProvider
+  class OfferSerializer < ActiveModel::Serializer
+    attribute :id, key: '@id'
+    attribute :type, key: '@type'
+    attribute :offeres_to, key: 'dfc:offeres_to'
+    attribute :price, key: 'dfc:price'
+    attribute :stock_limitation, key: 'dfc:stockLimitation'
+
+    def id
+      "/offers/#{object.id}"
+    end
+
+    def type
+      'dfc:Offer'
+    end
+
+    def offeres_to
+      {
+        '@type' => '@id',
+        '@id' => nil
+      }
+    end
+
+    def stock_limitation
+      object.on_hand
```

Comment:
I'm a bit concerned we're displaying stock levels of variants in lots of places but there's no variant override scoping anywhere...

Reply:
I don't fully understand the use cases and context, but I'm also concerned we're showing product lists in general with no reference to order cycles, tags, whether or not the enterprise is considered "active", and various other things that would all modify product lists.

Reply:
I think here we are not looking at OCs/availability.
We are simply looking at catalog without any OCs/overrides. We will need to include that in Offers at some point!

Reply:
We can discuss about data mapping and change it, no issue about that, I needed to return a value and I thought it was the closest fit.

Reply:
yes, this is the closest fit.
As long as we are on catalog looking at products and variants only, there's no problem in ignoring variant overrides.
As soon as you start looking at orders, you must include variant overrides otherwise data will be wrong.
A variant in a line item needs to be scoped to the distributor of the order, otherwise you will be looking at data (stock and price) from the supplier not the distributor.